src/compiler/compiler.ml

```diff
@@ -347,7 +347,7 @@ let finalize_typing ctx tctx =
 
 let filter ctx tctx ectx before_destruction =
 	Timer.time ctx.timer_ctx ["filters"] (fun () ->
-		DeprecationCheck.run ctx.com;
+		DeprecationCheck.run ctx.com (List.filter (fun t -> not (FiltersCommon.is_cached ctx.com t)) ctx.com.types);
```

Comment:
Can't we call this from the `run` function in filters.ml which already determines the `new_types`?